eng/pipelines/templates/pipeline.yml

```diff
@@ -148,8 +218,107 @@ stages:
       BuildRTM: "true"
     pool:
       name: VSEngSS-MicroBuild2022-1ES
+    templateContext:
+      mb:
+        localization:
+          enabled: true
+        signing:
+          enabled: true
+          signType: "$(SigningType)"
+        swix:
+          enabled: true
+        optprof:
+          enabled: "and(succeeded(), eq(variables['BuildRTM'], 'false'), eq(variables['IsOfficialBuild'], 'true'))"
+          OptimizationInputsLookupMethod: DropPrefix
+          DropNamePrefix: OptimizationInputs/$(System.TeamProject)/$(Build.Repository.Name)
+          ShouldSkipOptimize: $(ShouldSkipOptimize)
+          AccessToken: $(System.AccessToken)
+      outputs:
+      - output: pipelineArtifact
+        displayName: 'Publish buildinfo.json as an artifact'
+        condition: "and(succeeded(), eq(variables['BuildRTM'], 'false'))"
+        targetPath: '$(Build.Repository.LocalPath)\artifacts\buildinfo.json'
+        artifactName: 'BuildInfo'
+
+      - output: pipelineArtifact
+        displayName: 'Publish NuGet.CommandLine.Test as artifact'
```

Comment:
We're publishing this artifact in 2 different phases, is that intentional/necessary?

Reply:
Good catch, nope, the BuildRTM condition will prevent it so it's not needed. I removed several instances of this.